pywhy_stats/fisherz.py

```diff
@@ -0,0 +1,129 @@
+"""Independence test using Fisher-Z's test.
+
+This test is also known as the partial correlation independence test.
+It works on Gaussian random variables.
+
+When the data is not Gaussian, this test is not valid. In this case, we recommend
+using the Kernel independence test at <insert link>.
+
+Examples
+--------
+>>> import pywhy_stats as ps
+>>> res = ps.fisherz.ind([1, 2, 3], [4, 5, 6])
+>>> print(res.pvalue)
+>>> 1.0
+"""
+
+from math import log, sqrt
+from typing import Optional
+
+import numpy as np
+from numpy.typing import ArrayLike
+from scipy.stats import norm
+
+from .pvalue_result import PValueResult
+
+
+def ind(X: ArrayLike, Y: ArrayLike, correlation_matrix: Optional[ArrayLike] = None) -> PValueResult:
+    """Perform an independence test using Fisher-Z's test.
+
+    Works on Gaussian random variables. This test is also known as the
+    correlation test.
+
+    Parameters
+    ----------
+    X : ArrayLike of shape (n_samples,)
+        The first node variable.
+    Y : ArrayLike of shape (n_samples,)
+        The second node variable.
+    correlation_matrix : ArrayLike of shape (2, 2), optional
+        The precomputed correlation matrix between X and Y., by default None.
+
+    Returns
+    -------
+    X : float
```

Comment:
Maybe let's call this `statistic` or more explicit `test_statistic` instead of `X`? Otherwise, it's a bit confusing with the input `X`.